apps/explorer/src/app/components/links/asset-link/asset-link.tsx

```diff
@@ -5,27 +5,35 @@ import {
   useAssetDataProvider,
   useAssetDetailsDialogStore,
 } from '@vegaprotocol/assets';
+import { useNavigate } from 'react-router-dom';
+import { Routes } from '../../../routes/route-names';
 
 export type AssetLinkProps = Partial<ComponentProps<typeof ButtonLink>> & {
   assetId: string;
+  asDialog?: boolean;
 };
 
 /**
  * Given an asset ID, it will fetch the asset name and show that,
  * with a link to the assets modal. If the name does not come back
  * it will use the ID instead.
  */
-export const AssetLink = ({ assetId, ...props }: AssetLinkProps) => {
+export const AssetLink = ({ assetId, asDialog, ...props }: AssetLinkProps) => {
   const { data: asset } = useAssetDataProvider(assetId);
 
   const open = useAssetDetailsDialogStore((state) => state.open);
+  const navigate = useNavigate();
   const label = asset?.name ? asset.name : assetId;
   return (
     <ButtonLink
       data-testid="asset-link"
       disabled={!asset}
       onClick={(e) => {
-        open(assetId, e.target as HTMLElement);
+        if (asDialog) {
```

Comment:
**praise  (non-blocking)**: When I asked if this PR should update the `AssetLink` I assumed you'd just make it always link to the page. This option is much better 🥳

I've really liked on some pages that it can just pop up. I don't have a rule of thumb in mind for when a dialog is better than a link... so this is perfect.